proposals/graphql/api/src/main/java/org/eclipse/microprofile/graphql/Id.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.graphql;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marks a field as a Id Scalar Type.
+ * 
+ * <br><br>
+ * For example, a user might annotate a method's parameter as such:
+ * <pre>
+ * public class Person {
+ *      {@literal @}Id
+ *      private int id;
+ *      private String name;
+ *      private String surname;
+ * 
+ *      // Getters and Setters
+ * }
+ * </pre>
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.PARAMETER, ElementType.TYPE_USE, ElementType.FIELD, ElementType.TYPE})
```

Comment:
More for my education than a code review comment, but what cases would you put `@Id` on a type?

Reply:
To create a GraphQL Id scalar type in the schema. There is no java type of ID, hence the annotation.